src/cmd-buildextend-azure

```diff
@@ -0,0 +1,155 @@
+        else:
+            azure_nv = f'{base_name}-{args[0].build}'
+        # Used in uploading
+        self.azure_vhd_name = f'{azure_nv}-azure.vhd'
+
+        img_qemu = os.path.join(
+            self.build_root, self.meta['images']['qemu']['path'])
+        tmp_img_azure = os.path.join(TMPDIR, (azure_nv + '.qcow2'))
+        tmp_img_azure_vhd = os.path.join(TMPDIR, self.azure_vhd_name)
+        run_verbose([f"{cosa_dir}/gf-platformid",
+                     img_qemu, tmp_img_azure, "azure"])
+        run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'vhd',
+                    tmp_img_azure, tmp_img_azure_vhd])
+        os.unlink(tmp_img_azure)
+
+        fsize = subprocess.check_output([
+            "stat", "--format", '%s', tmp_img_azure_vhd])
+        log.debug(" * calculating checksum")
+        self._found_files[tmp_img_azure_vhd] = {
+                "local_path": os.path.abspath(tmp_img_azure_vhd),
+                "path": os.path.basename(tmp_img_azure_vhd),
+                "md5": md5sum_file(tmp_img_azure_vhd),
+                "size": int(fsize)
+        }
+        log.debug(" * size is %s", self._found_files[tmp_img_azure_vhd]["size"])
+        log.debug(" * md5 is %s", self._found_files[tmp_img_azure_vhd]["md5"])
+
+
+def cli():
+    """
+    Parse args and dispatch
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--azure-profile", help="Azure Profile json file.", required=True)
+    parser.add_argument("--azure-subscription", help="Azure subscription name. If unset, the first is used",
+                        required=False)
+    parser.add_argument("--container", help="Azure blob container.", required=True)
+    parser.add_argument("--force", help="Replace existing images and upload",
+                        action="store_true",
+                        default=bool(os.environ.get("AZURE_FORCE", False)))
+    parser.add_argument("--name-suffix", help="Append suffix to name",
+                        required=False)
+    parser.add_argument("--storage_account", help="Azure storage account.", required=True)
+
+    args = parser.parse_args()
+
+    # Identify the builds
+    build = Build(os.path.join('builds', args.build), args.build)
+
+    if 'azure' in build.meta['images'] and not args.force:
+        print("Azure image already built; use --force to rebuild")
+        sys.exit(0)
+
+    build.build_artifacts()
+    run_ore(args, build)
+
+
+def run_ore(args, build):
+    """
+    Execute ore to upload a Azure compatible VHD with corresponding metadata
+
+    :param args: The command line arguments
+    :type args: argparse.Namespace
+    :param build: Build instance to use
+    :type build: Build
+    """
+    vhd_file = f"{build.build_root}/{build.azure_vhd_name}"
+    tmp_img_azure_vhd = build.get_artifacts()[0]["local_path"]
+    base_name = build.meta['name']
+    # upload-blob storage-account container blob-name file
+    ore_args = ['ore', 'azure',
+                'upload-blob',
```

Comment:
Per discussion we also need `create-image` so it's bootable right?

Reply:
Yes, probably also `replicate-image` and `share-image` (although as optional flags maybe?) to match AWS